src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Tooltip/DefaultLSPTagHelperTooltipFactory.cs

```diff
@@ -51,6 +51,17 @@ internal class DefaultLSPTagHelperTooltipFactory(ISnapshotResolver snapshotResol
 
             var tagHelperType = descriptionInfo.TagHelperTypeName;
             var reducedTypeName = ReduceTypeName(tagHelperType);
+
+            // If the reducedTypeName != tagHelperType, then the type is prefixed by a namespace
+            if (reducedTypeName != tagHelperType)
+            {
+                var removeStartIndex = tagHelperType.Length - reducedTypeName.Length;
+                var TypeNamespace = tagHelperType.Remove(removeStartIndex, reducedTypeName.Length);
```

Comment:
Nit, locals should start with a lowercase letter:
```suggestion
                var typeNamespace = tagHelperType.Remove(removeStartIndex, reducedTypeName.Length);
```


Though also, you could consider just `descriptionBuilder.Append(tagHelperType[..^reducedTypeName.Length]);` as the entire body of this if statement.

Reply:
Thank you for teaching me how to use the range operator in C#

Reply:
I had to teach myself first, I still am not used to it. Before I commented I opened sharplab to confirm what I was suggesting was actually going to work :)